### PR TITLE
Archunit 0.23.0

### PR DESCRIPTION
**Pull request by timtebeek**
Closes #1815.

Method references are now picked up, leading to fewer false positives in detecting unused methods.
Required a minor code change to now use `JavaCodeUnitAccess`.

Updated store to only freeze the for now disabled unused methods test, as the unused public methods test now no longer has a false positive. We can reconsider whether that test needs to remain disabled, as there should no longer be false positives.

Ideally and eventually we can remove true unused methods through #1702; until then this minimal change allows us to pick up new releases of ArchUnit.

### CODE REVIEW

**Comment by jamietanna** [comment]
I'll try and have a look today, but @tomakehurst will still need to do the honours of merging as I've not yet got commit access ☺

**Comment by jamietanna** [comment]
I think this LGTM - happy for Tom to have another pair of eyes and merge when ready :+1:

**Comment by timtebeek** [comment]
Discussed outside GitHub; this one is good to go it seems! :)

**Review by jamietanna — APPROVED** [review]

File: src/test/java/com/github/tomakehurst/wiremock/archunit/UnusedCodeTest.java
Comment (tomakehurst) [inline src/test/java/com/github/tomakehurst/wiremock/archunit/UnusedCodeTest.java:166]: @timtebeek this message change still implies that we're failing on any unused public methods. Am I misunderstanding?

Comment (timtebeek) [reply src/test/java/com/github/tomakehurst/wiremock/archunit/UnusedCodeTest.java:166]: That's merely a result of the git diff rendering weirdly in the presence of indentation changes.
The `nonPublicMethodsShouldNotBeUnusedFrozen` isn't really new; it was there before already.
But because the `freeze` was moved from that test to `methodsShouldNotBeUnused` the diff is off.

There's nothing new in terms of what's enforced in this MR; all as it was, with `methodsShouldNotBeUnused` disabled for now. The (now) empty freeze store for `nonPublicMethodsShouldNotBeUnusedFrozen` was repurposed to serve `methodsShouldNotBeUnused`. Hence any confusion; but nothing new is enforced just yet.

Comment (tomakehurst) [reply src/test/java/com/github/tomakehurst/wiremock/archunit/UnusedCodeTest.java:166]: OK, got it (I think!). I'll merge.

Comment (timtebeek) [reply src/test/java/com/github/tomakehurst/wiremock/archunit/UnusedCodeTest.java:166]: Thanks! Glad to see this merged :)

**Pull request merged**
